### ISSUE TITLE
Concept name in constrained-parameter gets wrong semantic coloring

### ISSUE THREAD

**Issue by HighCommander4** [clangd/clangd#464]
```c++
template <typename T>
concept Concept = true;

template <Concept C>  // <--
void foo();
```

On the indicated line, the `Concept` token is colored as a "primitive" rather than as a concept.

**Comment by HighCommander4** [comment]
A related case: in a _partial-concept-id_, the concept name does not get a semantic coloring at all:

```
template <typename T, typename U>
concept Concept = true;

template <Concept<int> C>  // <--
void foo();
```

**Comment by HighCommander4** [comment]
[Proposed fix](https://reviews.llvm.org/D84136)

**Comment by sam-mccall** [comment]
Fixed in 00d7b7d014f90aaaacaef6f9c778614b09356bf0 I think!
